### DISCUSSION TITLE
AGridder Monitor

### DISCUSSION THREAD

**Discussion by shittymodernart** [apohl79/audiogridder#1046] [Q&A]
How do I interpret the values in shown for "Perf Proc" and "Perf Net" in AGridder? I had assumed that Perf Proc was the amount of processing latency and Perf Net was the network latency, but when I changed the oversampling to 16x on Pro-L 2 (the bottom plugin chain) my Perf Net speed went from 7.59ms to around 20ms, but the Perf Proc didn't change. 

I'm assuming that "Buf" is the amount of blocks that chain is buffered at. What is "Bav?"

<img width="633" alt="Screen Shot 2022-10-05 at 9 59 09 AM" src="https://user-images.githubusercontent.com/114954191/194079994-09416f48-741f-48b9-94b2-ba4783c7e7ae.png">

**Comment by apohl79** [comment]
https://github.com/apohl79/audiogridder/discussions/1017#discussioncomment-3634547

**Reply by shittymodernart** [reply]
> Perf Proc = the time spent in the DAW audio thread for processing a buffer of audio passed from the DAW to AG (avg over the last 32 calls)
Perf Net = the time spent sending and receiving one block of audio to/from the server (avg over the last 32 blocks)

Why would increasing the processing load through oversampling increase the time spent sending and receiving, but not the time spent in the DAW audio thread for processing?

**Reply by apohl79** [reply]
It is the time sending from the plugin to the server + the time the server needs to process the audio through the plugins + the time for sending it back to the plugin.

**Comment by apohl79** [comment]
As long as your network latency does not go red (for a "longer" period of time), the read buffer queue of a plugin should not go empty and block the audio thread. So Perf Proc should not be affected.

**Reply by shittymodernart** [reply]
And if it does start being effected, the solution is to just increase the buffer size for that plugin chain?

**Reply by apohl79** [reply]
In theory: if it is just temporary spikes, a bigger buffer might be sufficient. But if the plugin chain is generally performing slower then the time available in the realtime window (this depends on your sample rate and sample block size, for example 48K, 1024 sample blocks -> 21ms), the buffer will probably always run empty at some point, as the data is just not coming back fast enough.

In the end it is trial and error. Some DAWs might wait longer and compensate, others might run into errors, drop audio or stop. You could try increase the block size (if the DAW allows for this). You could also try to split a chain into multiple AG plugins. This seems to work for some people.

**Reply by shittymodernart** [reply]
What does the manual delay setting do? I started experiencing a delay today in a project with all AG plugins set to a 2-block buffer, so I switched it to no buffer and I still had the delay for some reason. I tried adding a negative manual delay to compensate but I think that made it worse. Adding a positive manual delay fixed the issue for me. So I guess I know what it does, but my question is more specifically what is it's intended purpose?

**Reply by apohl79** [reply]
It was just added as a workaround for an instrument that was causing PDC issues in logic. This is just changing the sample latency value AG reports to the DAW. Under normal circumstances you would not need to touch it.
